src/js/shop/components/Checkout/Address.js

```diff
@@ -0,0 +1,197 @@
+        (!deliveryAddressIsBillingAddress ? EMPTY_ADDRESS : null);
+
+    return (
+        <form onSubmit={onSubmit}>
+            <div className="row">
+                {/*Personal details*/}
+                <div className="col-xs-12 col-md-6">
+                    <h3 className="checkout__title">
+                        <FormattedMessage
+                            description="Checkout address: personal details"
+                            defaultMessage="Personal details"
+                        />
+                    </h3>
+
+                    <PersonalDetailsFields
+                        prefix="customer"
+                        firstName={customer.firstName}
+                        lastName={customer.lastName}
+                        email={customer.email}
+                        phone={customer.phone}
+                        onChange={onChange}
+                    />
+                </div>
+            </div>
+
+            <div className="row">
+                {/*Delivery address*/}
+                <div className="col-md-6 col-xs-12">
+                    <h3 className="checkout__title">
+                        <FormattedMessage
+                            description="Delivery address: deliveryAddress"
+                            defaultMessage="Delivery address"
+                        />
+                    </h3>
+
+                    <AddressFields
+                        prefix="deliveryAddress"
+                        company={deliveryAddress.company}
+                        chamberOfCommerce={deliveryAddress.chamberOfCommerce}
+                        street={deliveryAddress.street}
+                        number={deliveryAddress.number}
+                        city={deliveryAddress.city}
+                        postalCode={deliveryAddress.postalCode}
+                        country={{
+                            value: deliveryAddress.country,
+                            label: SUPPORTED_COUNTRIES[deliveryAddress.country],
+                        }}
+                        onChange={onChange}
+                    />
+
+                    <div className="form-check checkbox-flex">
+                        <input
+                            type="checkbox"
+                            className="form-check-input"
+                            id="deliveryAddressIsBillingAddress"
+                            checked={deliveryAddressIsBillingAddress}
+                            onChange={() => {
+                                setDeliveryAddressIsBillingAddress(
+                                    !deliveryAddressIsBillingAddress
+                                );
+                                onChange({
+                                    target: {
+                                        name: "billingAddress",
+                                        value: null,
+                                    },
+                                });
+                            }}
```

Comment:
suggested code unfortunately doesn't work because React is kinda weird with checkboxes. Additionally, the `value: null` is on purpose to reset the billing address data when the checkbox changes to prevent accidentally recording data that we don't need.